Homework_2/report/main.tex

```diff
@@ -0,0 +1,173 @@
+
+Tekom postopka implementacije funkcije \texttt{simpson2D(n)} smo implementirali funkcije \texttt{simpson6D(n)}, ki je naivna implementacija 6-kratnega integrala, \texttt{simpson3D(n)}, ki je implementacija 3-kratnega integrala in \texttt{monte\_carlo\_integration(n)}, ki izračuna 6-kratni integral z Monte Carlo metodo.
+Te funkcije smo implementirali, saj smo sprva mislili, da bo ne bo potrebna redukcija 6-kratnega integrala v 3-kratnega in 2-kratnega. Sedaj te funkcije lahko uporabimo za primerjavo rezultatov.
+
+\section{Rezultati}
+Da dosežemo željeno natančnost, povečujemo $n$ in preverimo da je absolutna vrednost kvocienta razlike med dvema zaporednima izračunoma in vrednosti zadnjega izračuna manjša od $10^{-10}$. Pri $n=200000$ dobimo silo, zaokroženo na 10 decimalk, $F\approx-0.2479229692$. Minus označuje, da je sila med kockama privlačna.
+Pri tem je relativna natančnost $6.76 \cdot 10^{-11}$, kar je manj kot $10^{-10}$, zato lahko rezultat sprejmemo. Čas izvajanja je bil približno $1500$ sekund.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.75\textwidth]{figures/force_convergence.png}
+    \caption{Konvergenca različnih metod.}
+    \label{fig:convergence}
+\end{figure}
+Za konec še primerjamo konvergenco in časovno zahtevnost rezultatov različnih metod na različnih intervalih za $n$. Na sliki \ref{fig:convergence} so prikazane vrednosti sile za različne $n$.
+Kljub temu, da je metoda \texttt{simpson6D} najpočasnejša, izgleda, da ima najboljšo konvergenco. \texttt{simpson3D} in \texttt{simpson2D} imata bistveno večja odstopanja pri $n=2,6,10,14,18$, kar je verjetno posledica tega, da zaradi preoblikovanja rešujemo drugačen integral kot na začetku. Ima pa \texttt{simpson2D} boljše rezultate kot \texttt{simpson3D}, kar je pričakovano,
+saj je integral po $z$ že bil izračunan analitično. \texttt{monte\_carlo\_integration} je najslabša metoda, saj je nedeterministična in je odvisna od naključne izbire točk.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.75\textwidth]{figures/force_time.png}
+    \caption{Čas izvajanja različnih metod.}
+    \label{fig:time}
+\end{figure}
+
+Na sliki \ref{fig:time} so prikazani časi izvajanja različnih metod. Pričakovano je \texttt{simpson6D(n)} najpočasnejša. Metoda \texttt{simpson3D(n)} je bistveno hitrejša, vendar je \texttt{simpson2D(n)} najhitrejša od determinističnih metod. Metoda \texttt{monte\_carlo\_integration(n)} je hitrejša od vseh, kar je pričakovano, saj je potrebno izračunati le $n$ naključnih točk in izračunati povprečno vrednost integranda, torej ima časovno kompleksnost $O(n)$.
+Ker pa je ta metoda nedeterministična, v nalogi pa iščemo natančno vrednost na 10 decimalk, te metode nismo uporabili. Če bi zadoščalo podati rezultat na 10 decimalk natančno z intervalom zaupanja, bi morda uporabili to metodo, odvisno od tega kako velik interval zaupanja bi želeli.
```

Comment:
Vse metode, ki jih uporabljamo so približne. 
To, da je metoda nedeterministična, ne pomeni, da je ne moremo uporabiti za izračun z določeno natančnostjo. Napako lahko ocenimo s centralnim limitnim izrekom. Pri tem obstaja majhna verjetnost, da željena natančnost ni dosežena. Prav tako ocene za napako, ki jih dobimo tako, da primerjamo rezultate za različne n-je niso 100% zanesljive.